### PR TITLE
fix #148 and #162

### LINKED ISSUE #148: Interaction with autodoc plugin not working ?
I'm using the autodoc sphinx plugin, so in order to work on the documentation from the docstrings in my code I would expect the following command to work:
```sh
sphinx-autobuild docs build --watch src
```

It seems to partially work, in the sense that editing a source file and saving indeed triggers a rebuild, but the output does not change. From my experimentation, it seems like the code is somehow cached, and this it's always the version existing at the first launch of the above mentioned command that gets built.

I haven't found a solution so far, is this a bug ? a configuration problem ?

### PR DESCRIPTION
**Pull request by return42**
See commit messages for more details

### CODE REVIEW

File: sphinx_autobuild/build.py [hunk 1]
```diff
@@ -31,13 +28,11 @@ def __call__(self, *, rebuild: bool = True):
...
 
-        show(command=["sphinx-build"] + self.sphinx_args)
-
-        # NOTE:
-        # sphinx.cmd.build.build_main is not considered to be public API,
-        # but as this is a first-party project, we can cheat a little bit.
-        if return_code := build_main(self.sphinx_args):
-            print(f"Sphinx exited with exit code: {return_code}")
+        try:
+            show(command=["sphinx-build"] + self.sphinx_args)
```
Comment (AA-Turner) [inline sphinx_autobuild/build.py:32]: This doesn’t need to be inside the try block.

Comment (return42) [reply sphinx_autobuild/build.py]: @AA-Turner fixed it, see [diff](https://github.com/sphinx-doc/sphinx-autobuild/compare/f440a7415d7f5e58bac22ec9b549abdddff2e210..050a5f6bb8a96fb1807bb0efee2317c4f2668e75)

**Comment by AA-Turner** [comment]
Please avoid force pushing; all PRs will be squash-merged.

A

File: sphinx_autobuild/__main__.py [hunk 2]
```diff
@@ -27,7 +27,7 @@ def main(argv=()):
     """Actual application logic."""
     colorama.init()
 
-    args, build_args = _parse_args(list(argv))
+    args, build_args = _parse_args(list(argv) or sys.argv[1:])
```
Comment (AA-Turner) [inline sphinx_autobuild/__main__.py:30]:
```suggestion
    if not argv:
        # entry point functions don't receive args
        argv = sys.argv[1:]

    args, build_args = _parse_args(list(argv))
```

**Pull request merged**